Langchain intro - Example - example_selectors fix + Vertex AI MatchingEngine GCS url fix for index creation

arindam-b opened the pull request:
Fix - Vertex AI Matching engine requires GCS url for init_index. The GCS URL was wrong which is corrected and a small typo.


Fix - embedded links in the notebook for example_selectors, text splitter, summarization chain and chain creation in Langchain intro.

polong-lin commented:
Hi @arindam-b , thanks for fixing these links! Before we can proceed with merging, you would need to accept and sign the CLA: 
https://github.com/GoogleCloudPlatform/generative-ai/pull/104/checks?check_run_id=14908796396

arindam-b commented:
Hi @polong-lin, I have accepted and signed the CLA, thanks.

polong-lin commented on language/examples/langchain-intro/intro_langchain_palm_api.ipynb:
```diff
@@ -1025,9 +1025,9 @@
     "### **Text Splitters**\n",
     "\n",
-    "[Text Splitters](https://python.langchain.com/en/latest/modules/indexes/text_splitters.html) are a way to deal with input token limits of LLMs by splitting text into chunks.\n",
+    "[Text Splitters](https://python.langchain.com/docs/modules/data_connection/document_transformers/) are a way to deal with input token limits of LLMs by splitting text into chunks.\n",
```
Looks like this link for Document Transformers doesn't match the text: "Text Splitters"?

arindam-b replied:
Thanks for the comment. If you look at the langchain documents, earlier "Indexes" were part of the Modules and "Text Splitter" was under "Indexes". Now, the structure is: "Modules" -> "Data connection" -> "Document transformers" -> "Text splitters". So the link I have specified describes the Text splitters only, which falls under Transformers. You can find the image below.

![screen-shot](https://github.com/GoogleCloudPlatform/generative-ai/assets/59955214/a87d8e0a-8962-4799-a7cc-db92c516ad08)


It is also worth noting, the langchain picture in the notebook with different core components needs to be updated, for which I will raise a new PR. Let me know your comments.

arindam-b commented:
@polong-lin: Another modification was added, this time to the Vertex AI Matching engine index creation. That's why I have modified the PR title and added some details. Could you please review it? Thanks.

polong-lin reviewed (approved):
LGTM -- thank you for spotting these issues, and fixing the code/links!
